src/System.Memory.Polyfill/System/Parse.cs

```diff
@@ -0,0 +1,18 @@
+// Licensed to the .NET Foundation under one or more agreements.
+// The .NET Foundation licenses this file to you under the MIT license.
+// See the LICENSE file in the project root for more information.
+
+namespace System
+{
+    public static partial class Int32Polyfill
+    {
+        public static bool TryParse(ReadOnlySpan<char> buffer, out int value)
+        {
+#if NETCOREAPP2_1
+            return int.TryParse(buffer, out value);
```

Comment:
You may want to avoid using the versioned define as it makes it more difficult if you decide to retarget.  Instead use `#if NETCOREAPP`

Reply:
But this api exists only in netcoreapp 2.1

Reply:
You aren't defining targeting here, you're just ifdefing the code.  Imagine that you wanted to cross-compile for netcoreapp2.2 as well in the future, you'd need to touch this ifdef for no reason.  I think you're more likely to cross-compile for a future version of netcoreapp than you are an older version that doesn't have this API.
Alternatively you can define your own feature flag that represents what you want and make sure that's defined as appropriate.

Reply:
Yeah, I agree it would work today, but it won't if we ever add new targets. Correct?